ctapipe/tools/apply_models.py

```diff
@@ -118,6 +113,10 @@ class ApplyModels(Tool):
             },
             "Overwrite output file if it exists",
         ),
+        "disable-progress": (
```

Comment:
I think the flag definition should be left as before,  but the action needs to changed accordingly.

I think the desired behavior is:

* Flag not given sets `disable=None`, so tqdm checks if it runs in an interactive environment or not
* `--progress` forces the progress bar -> `disable=False`
* `--no-progress` forces the progress bar to be disabled -> `disable=True`.

This would be the case for:


```
"progress": ({"ApplyModels": {"disable_progress_bar": False}}, "Always enable progress bar."),
"no-progress": ({"ApplyModels": {"disable_progress_bar": True}}, "Always disable progress bar."),
```

Reply:
Thanks for the suggestion, indeed that would not change the current implementation. It is actually what we briefly discussed in the meeting, I think.

Reply:
Looking at all this duplicated code now, we could move these options to the base `Tool` class with a method `progress_bar` so we have this only in one place and the implementations can just do:

```
for foo in self.progress_bar(iterable, **options):
    ...
```